src/behavior.cpp

```diff
@@ -0,0 +1,90 @@
+#include "behavior.h"
+
+using namespace behavior;
+
+namespace behavior
+{
+sequential default_sequential;
+fallback default_fallback;
+}
+
+std::pair<status, const node *> sequential::evaluate( const Creature *subject,
```

Comment:
`std::pair<status, const node *>` could be converted into something shorter with using.  I think it would make the code slightly more readable.

Reply:
`Using`s tend to obfuscate the code, sometimes making it harder to read rather than easier. What could probably improve readability of the return values, is a `struct`. Something like:
```c++
struct outcome_t {  // or whatever.
    status_t status;
    const node_t *node;
};
```
Pros:
- `outcome.status` and `outcome.node` are more expressive than `outcome.first` and `outcome.second` respectively.
- Easier to extend if needed.

Reply:
I'm not a fan of ``using`` for creating aliases, a struct instead of a pair is an option though.